### PR TITLE
[SPARK-45742][CORE][CONNECT][MLLIB][PYTHON] Introduce an implicit function for Scala Array to wrap into `immutable.ArraySeq`.

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
Currently, we need to use `immutable.ArraySeq.unsafeWrapArray(array)` to wrap an Array into an `immutable.ArraySeq`, which makes the code look bloated.

So this PR introduces an implicit function `toImmutableArraySeq` to make it easier for Scala Array to be wrapped into `immutable.ArraySeq`.

After this pr, we can use the following way to wrap an array into an `immutable.ArraySeq`:

```scala
import org.apache.spark.util.ArrayImplicits._

val dataArray = ...
val immutableArraySeq = dataArray.toImmutableArraySeq
```

At the same time, this pr replaces the existing use of `immutable.ArraySeq.unsafeWrapArray(array)` with the new method. 

On the other hand, this implicit function will be conducive to the progress of work SPARK-45686 and SPARK-45687.

### Why are the changes needed?
Makes the code for wrapping a Scala Array into an `immutable.ArraySeq` look less bloated.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: common/utils/src/main/scala/org/apache/spark/util/ArrayImplicits.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ */
+private[spark] object ArrayImplicits {
+
+  implicit class SparkArrayOps[T](xs: Array[T]) {
+
+    /**
+     * Wraps an Array[T] as an immutable.ArraySeq[T] without copying.
+     */
+    def toImmutableArraySeq: immutable.ArraySeq[T] =
+      if (xs eq null) null
```
Comment (LuciferYang) [inline common/utils/src/main/scala/org/apache/spark/util/ArrayImplicits.scala:33]:
```scala
Welcome to Scala 2.13.12 (OpenJDK 64-Bit Server VM, Java 17.0.8).
Type in expressions for evaluation. Or try :help.

scala> val a: Array[Int] = null
val a: Array[Int] = null

scala> scala.collection.immutable.ArraySeq.unsafeWrapArray(a)
val res0: scala.collection.immutable.ArraySeq[Int] = null

```

File: core/src/test/scala/org/apache/spark/util/ArrayImplicitsSuite.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ */
+
+package org.apache.spark.util
+
+import scala.collection.immutable
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.util.ArrayImplicits._
+
+class ArrayImplicitsSuite extends SparkFunSuite {
```
Comment (LuciferYang) [inline core/src/test/scala/org/apache/spark/util/ArrayImplicitsSuite.scala:25]: Although this pr has modified the `utils` module, looking at the test parameters `-Phadoop-3 repl/test avro/test examples/test`, it did not execute the tests of the utils module, so it seems that the test of the utils module has not been actually verified on GA,  put it in the core module for now.

The issue with the utils module test will be followed up separately.

**Comment by LuciferYang** [comment]
Could you help review this pr if you have time. @dongjoon-hyun Thanks ~

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @LuciferYang .

**Comment by dongjoon-hyun** [comment]
Merged to master.

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun ~

**Pull request closed without merging**